bin/internal/update_dart_sdk.ps1

```diff
@@ -48,6 +48,12 @@ if ($engineRealm) {
     $dartSdkBaseUrl = "$dartSdkBaseUrl/$engineRealm"
 }
 $dartZipName = "dart-sdk-windows-x64.zip"
+# For ARM64, it's important to run native dart, as it will allow to have
+# PROCESSOR_ARCHITECTURE environment variable set correctly. If running x64
+# version, PROCESSOR_ARCHITECTURE is always set to AMD64 and can't be overriden.
+if ($env:PROCESSOR_ARCHITECTURE -eq "ARM64") {
+    $dartZipName = "dart-sdk-windows-arm64.zip"
+}
```

Comment:
Nit: you should be able to use if expression here

```suggestion
# For ARM64, it's important to run native dart, as it will allow to have
# PROCESSOR_ARCHITECTURE environment variable set correctly. If running x64
# version, PROCESSOR_ARCHITECTURE is always set to AMD64 and can't be overriden.
$dartZipName = if ($env:PROCESSOR_ARCHITECTURE -ne "ARM64") {
    "dart-sdk-windows-x64.zip"
} else {
    "dart-sdk-windows-arm64.zip"
}
```